pkg/api/v1/atlascustomresource.go

```diff
@@ -0,0 +1,13 @@
+package v1
+
+import (
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1/status"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+//+k8s:deepcopy-gen=false
+// AtlasCustomResource is the interface common for all Atlas entities
+type AtlasCustomResource interface {
```

Comment:
Should we add static interface checks to make sure our types implement this interface?

Reply:
do you mean smth like 
```
var _ AtlasCustomResource = AtlasProject
```

Reply:
Yes, pretty much